frontend/src/components/UpdateModal.tsx

```diff
@@ -27,46 +54,69 @@ export const UpdateModal: FunctionComponent<{
 
   useEffect(() => {
     onOpen && onOpen();
+    // eslint-disable-next-line
   }, []);
 
   useEffect(() => {
     setRowData(changes);
   }, [changes]);
 
-  const autoGroupColumnDef = useMemo(() => {
-    return {
-      headerName: "Primary Id",
-      field: "primaryId",
-    };
-  }, []);
-
-  const [updateSamplesMutation, { data, loading, error }] =
-    useUpdateSamplesMutation();
+  const [updateSamplesMutation] = useUpdateSamplesMutation();
 
-  const handleSubmitUpdates = () => {
-    const changesForSubmit: ChangeForSubmit = {};
+  async function handleSubmitUpdates() {
+    const changesByPrimaryId: ChangesByPrimaryId = {};
     for (const c of changes) {
-      if (changesForSubmit[c.primaryId]) {
-        changesForSubmit[c.primaryId][c.fieldName] = c.newValue;
+      const { primaryId, fieldName, newValue } = c;
+      if (changesByPrimaryId[primaryId]) {
+        changesByPrimaryId[primaryId][fieldName] = newValue;
       } else {
-        changesForSubmit[c.primaryId] = { [c.fieldName]: c.newValue };
+        changesByPrimaryId[primaryId] = { [fieldName]: newValue };
+      }
+    }
+
+    if (setUserEmail) {
+      if (!changes.some((c) => c.fieldName === "billed")) return;
+
+      if (!userEmail) {
+        const logInUser = new Promise<string | null>((resolve) => {
+          window.addEventListener("message", handleLogin);
+          openLoginPopup();
+
+          async function handleLogin(event: MessageEvent) {
+            if (event.data === "success") {
+              const userEmail = await getUserEmail();
+              window.removeEventListener("message", handleLogin);
+              resolve(userEmail);
+            }
+          }
+        });
+
+        const userEmail = await logInUser;
+        if (userEmail) {
+          setUserEmail(userEmail);
+          populateBilledBy(changesByPrimaryId, userEmail);
+        } else {
+          return;
+        }
+      } else {
+        populateBilledBy(changesByPrimaryId, userEmail);
       }
     }
```

Comment:
This code block checks if the user's changes include the `billed` field. If so, we'll populate the `billedBy` field with their username. If the user is not logged in, we'll ask them to login first in order to get their username.